### ISSUE TITLE
[@angular-eslint/sort-ngmodule-metadata-arrays] Sorting with Internationalization support

### ISSUE THREAD

**Issue by jmeinlschmidt** [angular-eslint/angular-eslint#1098]
If my runtime environment language is set to something different than `en-US`, linter might throw some *false positive* errors, even though my variables are using ASCII characters only, btw JavaScript supports Unicode as well (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Grammar_and_types). This issue is quite difficult to debug. Interestingly, VSCode doesn't show any error, but linter output does. 

```JSON
{
  "rules": {
    "@angular-eslint/sort-ngmodule-metadata-arrays": "error",
  }
}
```

```TS
@NgModule({
imports: [
    // ...
    MatChipsModule,
    MatDatepickerModule,
    // ...
  ],
})
export class Test{ }
```

The example is in a valid order for environment language set to `en-US`, however not for example `cs-CZ`, where letter `D` precedes letter `Ch` (see https://unicode-org.github.io/cldr-staging/charts/latest/collation/cs.html). In current state, developers have no options to modify this behaviour, except for changing their environment, which gets later inherited to the `localCompare` function, which does the ordering.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `13.2.1` |
| `@typescript-eslint/parser`        | `5.26.0` |
| `ESLint`                           | `8.16.0` |
| `node`                             | `14.20.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1303.6
@angular-devkit/build-angular   13.3.6
@angular-devkit/core            13.3.6
@angular-devkit/schematics      13.3.6
@angular/cdk                    13.3.7
@angular/cli                    13.3.6
@angular/fire                   7.3.0
@angular/flex-layout            13.0.0-beta.38
@angular/material               13.3.7
@schematics/angular             13.3.6
ng-packagr                      13.0.8
rxjs                            7.4.0
typescript                      4.6.4
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.
